src/main/java/org/motechproject/prevac/service/impl/PrimeVaccinationScheduleServiceImpl.java

```diff
@@ -67,19 +71,19 @@ public PrimeVaccinationScheduleDto createOrUpdateWithDto(PrimeVaccinationSchedul
             return new PrimeVaccinationScheduleDto(updateVisitWithDto(primeVisit, screeningVisit, dto));
         }
 
+        Clinic clinic = clinicDataService.findByExactSiteId(siteId);
```

Comment:
change it back and set clinic in createScreeningVisitFromDto and createPrimeVacVisitFromDto using site id from subject